### PR TITLE
[show][config] cli support for firmware upgrade on Y-Cable

### PR DESCRIPTION
**Pull request by vdahiya12**
Summary:
This PR provides the support for adding CLI commands for activatie, download, upgrade firmware on the Y-cable
In particular these Cli commands are supported:
sudo config muxcable firmware download Ethernet0
sudo config muxcable firmware rollback Ethernet0
sudo config muxcable firmware download ~/AEC_WYOMING_B52Yb0_MS_0.6_20201218.bin Ethernet0

Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>

#### What I did
Added the support for firmware upgrade CLI on Y cable

#### How I did it
added the changes in sonic-utilities/show and sonic-utilities/config by changing the muxcable.py

```
admin@STR43-0101-0101-01LT0:/usr$ sudo config muxcable firmware download ~/AEC_WYOMING_B52Yb0_MS_0.6_20201218.bin Ethernet0
firmware download successful Ethernet0

admin@STR43-0101-0101-01LT0:~/credo_script_01_21$ sudo config muxcable firmware activate Ethernet0
firmware activate successful Ethernet0

admin@STR43-0101-0101-01LT0:~/credo_script_01_21$ sudo config muxcable firmware rollback Ethernet0
firmware rollback successful Ethernet0
```

#### How to verify it
Ran the command on a 7050cx3 arista switch

### CODE REVIEW

**Review by jleveque — CHANGES_REQUESTED** [review]
There is a lot of duplicate code in these new functions. Can you look into writing a helper function(s) so as to share common code?

File: config/muxcable.py [hunk 1]
```diff
@@ -631,3 +631,242 @@ def setswitchmode(state, port):
...
+
+                status = perform_download_firmware(physical_port, fwfile)
+
+                if status is not True:
+                    rc = False
+
+        if rc:
+            sys.exit(CONFIG_SUCCESSFUL)
+        else:
+            sys.exit(CONFIG_FAIL)
```
Comment (jleveque) [inline config/muxcable.py:769]: `rc` seems redundant as a bool. Suggest setting `rc = CONFIG_SUCCESSFUL` on line 750, `rc = CONFIG_FAIL` on line 764, and here simply call `sys.exit(rc)`.

Suggest making this change throughout the file.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: show/muxcable.py [hunk 2]
```diff
@@ -820,3 +820,196 @@ def switchmode(port):
         click.echo(tabulate(body, headers=headers))
         if rc == False:
             sys.exit(EXIT_FAIL)
+
+
+def create_result_dict(physical_port, mux_info_dict, result_dict):
```
Comment (vdahiya12) [inline show/muxcable.py:825]: @jleveque should we move all this to y_cable.py or let it remain here ? 
this is in regards to the newer resultant dict you suggested. 
For example ? 
{
    "version_nic_active": "0.6MS",
    "version_nic_inactive": "0.5MS",
    "version_nic_next": "0.6MS",
    "version_self_active": "0.5MS",
    "version_self_inactive": "0.6MS",
    "version_self_next": "0.6MS",
    "version_peer_active": "0.6MS",
    "version_peer_inactive": "0.6MS",
    "version_peer_next": "0.6MS",
}

File: config/muxcable.py (hunk 1 above)
Comment (vdahiya12) [reply config/muxcable.py]: fixed

File: show/muxcable.py (hunk 2 above)
Comment (vdahiya12) [reply show/muxcable.py]: refactored, new API's used here

File: config/muxcable.py [hunk 3]
```diff
@@ -730,13 +731,16 @@ def download(fwfile, port):
     if port is not None and port != "all":
 
         physical_port_list = []
-        get_physical_port_list(port, physical_port_list)
+        physical_port_list, asic_index = get_physical_port_list(port)
+        f = open("demofile2.txt", "w")
+        f.write("yes ")
+        f.write("{} ".format(len(physical_port_list)))
```
Comment (jleveque) [inline config/muxcable.py:737]: Why write to this file?

Comment (vdahiya12) [reply config/muxcable.py]: these were left-over debug statements, removed now

**Review by jleveque — APPROVED** [review]

**Comment by yxieca** [comment]
@vdahiya12 this change cannot be cherry-picked to 202012 branch cleanly. Can you find the dependent PR or create separate PR with this change for 202012 branch?

**Comment by vdahiya12** [comment]
opened a PR for 202012 branch
https://github.com/Azure/sonic-utilities/pull/1558

**Pull request merged**